src/rosdiscover/interpreter/interpreter.py

```diff
@@ -47,15 +48,15 @@ def summarise(self) -> SystemSummary:
         node_to_summary = {s.fullname: s for s in node_summaries}
         return SystemSummary(node_to_summary)
 
-    def launch(self, filename: str) -> None:
+    def launch(self, launch_obj: Launch) -> None:
```

Comment:
maybe rename `launch_obj` to `launch_description`?

Reply:
done